adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,44 @@
 #include "impl.h"
+
+#include <stack>
 #include <vector>
+#include <utility>
+#include <algorithm>
+
+namespace {
 
-const std::vector<std::pair<char, char>> brackets {
+using BracketsPair = std::pair<char, char>;
+using BracketsDictionary = std::vector<BracketsPair>;
+const BracketsDictionary brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+} // namespace
+
+bool isValid(const std::string& source) {
+
+    std::stack<char> s;
+    for (auto c : source) {
+        auto iter = std::find_if(brackets.cbegin(), brackets.cend(), [c](const BracketsPair& p) {
+            return p.first == c || p.second == c;
+        });
+
+        if (iter == brackets.end()) {
+            continue;
+        }
+
+        if (iter->first == c) {
+            s.push(c);
+            continue;
```

Comment:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es77-minimize-the-use-of-break-and-continue-in-loops